### PR TITLE
[SPARK-20258][Doc][SparkR] Fix SparkR logistic regression example in programming guide (did not converge) 

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?

SparkR logistic regression example did not converge in programming guide (for IRWLS). All estimates are essentially zero:

```
training2 <- read.df("data/mllib/sample_binary_classification_data.txt", source = "libsvm")
df_list2 <- randomSplit(training2, c(7,3), 2)
binomialDF <- df_list2[[1]]
binomialTestDF <- df_list2[[2]]
binomialGLM <- spark.glm(binomialDF, label ~ features, family = "binomial")

17/04/07 11:42:03 WARN WeightedLeastSquares: Cholesky solver failed due to singular covariance matrix. Retrying with Quasi-Newton solver.

> summary(binomialGLM)

Coefficients:
                 Estimate
(Intercept)    9.0255e+00
features_0     0.0000e+00
features_1     0.0000e+00
features_2     0.0000e+00
features_3     0.0000e+00
features_4     0.0000e+00
features_5     0.0000e+00
features_6     0.0000e+00
features_7     0.0000e+00
```

### CODE REVIEW

File: examples/src/main/r/ml/glm.R [hunk 1]
```diff
@@ -44,8 +44,9 @@ gaussianGLM2 <- glm(label ~ features, gaussianDF, family = "gaussian")
 summary(gaussianGLM2)
 
 # Fit a generalized linear model of family "binomial" with spark.glm
-training2 <- read.df("data/mllib/sample_binary_classification_data.txt", source = "libsvm")
```
Comment (felixcheung) [inline examples/src/main/r/ml/glm.R]: is this an issue with `binary_classification_data` data?

**Review by felixcheung — APPROVED** [review]
LGTM

**Comment by actuaryzhang** [comment]
@felixcheung 
Just noticed that the current example for logistic regression in the programming guide did not seem to be a good one. It did not converge using IRWLS, and Quasi-Newton yielded almost zero estimates for all coefficients.

File: examples/src/main/r/ml/glm.R [hunk 2]
```diff
@@ -44,8 +44,9 @@ gaussianGLM2 <- glm(label ~ features, gaussianDF, family = "gaussian")
...
-df_list2 <- randomSplit(training2, c(7,3), 2)
+training2 <- read.df("/data/mllib/sample_multiclass_classification_data.txt", source = "libsvm")
```
Comment (felixcheung) [inline examples/src/main/r/ml/glm.R]: actually, you might need to leave it as relative path, ie. not starting with `/`

File: examples/src/main/r/ml/glm.R (hunk 1 above)
Comment (actuaryzhang) [inline examples/src/main/r/ml/glm.R]: just a bad example.

File: examples/src/main/r/ml/glm.R (hunk 2 above)
Comment (actuaryzhang) [inline examples/src/main/r/ml/glm.R]: Thanks! copy paste error. Corrected now.

**Comment by felixcheung** [comment]
right, tests don't run example anyway...

**Comment by felixcheung** [comment]
merged to master.

**Pull request closed without merging**